packages/common/src/types/explore.ts

```diff
@@ -22,13 +22,15 @@ export type ExploreJoin = {
     hidden?: boolean;
     fields?: string[]; // Optional list of fields to include from the joined table
     always?: boolean; // Optional flag to always join the table
+    relationship?: 'one-to-many' | 'many-to-one' | 'one-to-one';
 };
 
 export type CompiledExploreJoin = Pick<
     ExploreJoin,
-    'table' | 'sqlOn' | 'type' | 'hidden' | 'always'
+    'table' | 'sqlOn' | 'type' | 'hidden' | 'always' | 'relationship'
 > & {
-    compiledSqlOn: string; // Sql on clause with template variables resolved
+    compiledSqlOn: string; // SQL on clause with template variables resolved
+    tablesReferences?: string[]; // Tables referenced in SQL. Optional, to keep it backwards compatible.
```

Comment:
New project deploys/compiles will have this new metadata about which tables are referenced in a join SQL. It should have no impact on current projects.